feat: support for new mongodbatlas_atlas_user and mongodbatlas_atlas_users data sources (#1432)
mongodb/terraform-provider-mongodbatlas: examples/atlas-user/user-datasource/README.md

# Example - MongoDB Atlas User Data Source

> This project aims to provide a very straight-forward example for using the Atlas User Data Source.

Reviewer: simplifying the verb form makes this more direct
```suggestion
This project provides a very straight-forward example for using the Atlas User Data Source.
```

Variables Required to be set:

> - `public_key`: Atlas public key

Reviewer 2: ```suggestion
- `public_key`: Atlas Programmatic API public key
```

> - `private_key`: Atlas  private key

Reviewer: nit pick: remove extra space
```suggestion
- `private_key`: Atlas private key
```

> - `private_key`: Atlas  private key

Reviewer 2: ```suggestion
- `private_key`: Atlas Programmatic API private key
```

> - `user_id`: User ID of the Atlas User that will be fetched

Reviewer: removes passive voice + future tense

also suggesting that we use the same verb here as we do for the other data source example. We use `return` mostly in the docs.

```suggestion
- `user_id`: User ID of the Atlas User to return
```

> - `username`: Username of the Atlas User that will be fetched

Reviewer: removes passive voice + future tense
```suggestion
- `username`: Username of the Atlas User to return
```

> Example shows the two possible way the data source can be used, either providing the `user_id` or `username` attribute.

Reviewer: subject / verb agreement fix + removing passive voice
```suggestion
This example shows the two ways that you can use the data source, either providing the `user_id` or `username` attribute.
```

> For additional documentation, you can reference to [MongoDB Atlas API - Get User By ID](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/MongoDB-Cloud-Users/operation/getUser) and [MongoDB Atlas API - Get User By Username](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/MongoDB-Cloud-Users/operation/getUserByUsername) respectively.

Reviewer: minor edit for concision
```suggestion
For additional documentation, see [MongoDB Atlas API - Get User By ID](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/MongoDB-Cloud-Users/operation/getUser) and [MongoDB Atlas API - Get User By Username](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/MongoDB-Cloud-Users/operation/getUserByUsername) respectively.
```

Reviewer: question: should these be pointing to the v2 spec at https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/ instead?

Author: yes makes sense since we are calling the v2 API.
